### PR TITLE
Jaeger: Fix calling of search query with the correct time range

### LINKED ISSUE #88156: Explore: Jaeger Query Timestamps are not sent correctly
### What happened?

What happened: The timestams sent to the Jaeger API is wrong
Expected: `2024-05-12 07:00:00 to 2024-05-12 12:00:00 UTC`

image

### What did you expect to happen?

Actual: `2024-05-21 21:11:41 to 2024-05-22 05:11:41 UTC`

### Did this work before?

I'm not sure. I only noticed after upgrading to Grafana 11

### How do we reproduce it?

1. Install a Jaeger Data source
2. Open Explore
3. Query a date that ends earlier than 6 hours ago

### Is the bug inside a dashboard panel?

This can also be reproduced in a dashboard, it is the same issue as Explore

### Environment (with versions)?

Grafana: 11.0.0
OS: Windows 10
Browser: Edge 124.0.2478.51 

### Grafana platform?

Docker

### Datasource(s)?

Jaeger
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by EgorKluch**
PR to fix https://github.com/grafana/grafana/issues/88156

### CODE REVIEW

**Comment by orloffv** [comment]
@fabrizio-grafana @joey-grafana Can you see this pr that fixes a critical bug?

**Comment by fabrizio-grafana** [comment]
@orloffv I'm not working on the Jaeger plugin anymore, but I'm checking who is the new owner.

**Comment by fabrizio-grafana** [comment]
Redirected to @grafana/oss-big-tent. Team, let me know if you need help with this!

File: public/app/plugins/datasource/jaeger/datasource.test.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ jest.mock('@grafana/runtime', () => ({
 }));
 
 describe('JaegerDatasource', () => {
+  const defaultSearchRangeParams = `start=${Number(defaultQuery.range.from) * 1000}&end=${Number(defaultQuery.range.to) * 1000}`;
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/jaeger/datasource.test.ts:42]: The tests are failing, you'll need to move `defaultQuery` up in the file. I'd suggest putting it just bellow `export const backendSr...`

File: public/app/plugins/datasource/jaeger/datasource.test.ts [hunk 2]
```diff
@@ -128,14 +130,28 @@ describe('JaegerDatasource', () => {
...
+    // @ts-ignore
+    query.range = undefined;
+
+    ds.query({ ...query, targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }] });
+
+    expect(mock).toHaveBeenCalledWith({
+      url: `${defaultSettings.url}/api/traces?service=jaeger-query&start=1704085200000000&end=1704106800000000&lookback=custom`,
+    });
+  });
+
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/jaeger/datasource.test.ts:154]:
```suggestion
  it('uses default range when no range is provided for search query,', async () => {
    const mock = setupFetchMock({ data: [testResponse] });
    const ds = new JaegerDatasource(defaultSettings);
    const query = {
      ...defaultQuery,
      targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }],
      // set range to undefined to test default range
      range: undefined,
    } as unknown as DataQueryRequest<JaegerQuery>;

    ds.query(query);
    expect(mock).toHaveBeenCalledWith({
      // Check that query has time range from 6 hours ago to now (default range)
      url: `${defaultSettings.url}/api/traces?service=jaeger-query&start=1704085200000000&end=2&lookback=custom`,
    });
  });
```

We shouldn't use `@ts-ignore` and we don't need to do stringify/parse the default query. We can just:

```
    const query = {
      ...defaultQuery,
      targets: [{ queryType: 'search', refId: 'a', service: 'jaeger-query', operation: ALL_OPERATIONS_KEY }],
      // Check that query has time range from 6 hours ago to now (default range)
      range: undefined,
    } as unknown as DataQueryRequest<JaegerQuery>;
```

**Review by ivanahuckova — CHANGES_REQUESTED** [review]
LGTM, but tests are failing so you'll need to fix that first. I have added some suggestions bellow.

**Review by ivanahuckova — APPROVED** [review]

**Comment by ivanahuckova** [comment]
Hi @EgorKluch! Thank you so much for pointing out this bug and creating fix. I went ahead and added a commit that fixes the failing test and merged the PR. It is going to be part of the next 11.2 release.

**Pull request merged**